core/src/test/java/org/apache/iceberg/util/TestTruncateUtil.java

```diff
@@ -0,0 +1,99 @@
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.util;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+class TestTruncateUtil {
+  @Test
+  public void testTruncateByteBuffer() throws Exception {
+    Assert.assertEquals(
+        "Should truncate binary longer than length",
+        ByteBuffer.wrap("abcd".getBytes("UTF-8")),
+        TruncateUtil.truncateByteBuffer(4, ByteBuffer.wrap("abcdefg".getBytes("UTF-8"))));
+    Assert.assertEquals(
+        "Should not pad binary shorter than length",
+        ByteBuffer.wrap("abc".getBytes("UTF-8")),
+        TruncateUtil.truncateByteBuffer(4, ByteBuffer.wrap("abc".getBytes("UTF-8"))));
+  }
+
+  @Test
+  public void testTruncateByte() {
+    Assert.assertEquals(
+        "Should return a byte value vs some other numeric type",
+        (byte) (Byte.MAX_VALUE - 1),
+        TruncateUtil.truncateByte(Byte.MAX_VALUE - 1, Byte.MAX_VALUE));
+  }
+
+  @Test
+  public void testTruncateShort() {
+    Assert.assertEquals(
+        "Should truncate a short and return a short",
+        (short) 0,
+        TruncateUtil.truncateShort(Short.MAX_VALUE + 1, Short.MAX_VALUE));
+  }
+
+  @Test
+  public void testTruncateInt() {
+    Assert.assertEquals(
+        "Should correctly truncate integer values and return an integer back",
+        Integer.MAX_VALUE - 1,
+        TruncateUtil.truncateInt(Integer.MAX_VALUE - 1, Integer.MAX_VALUE));
+  }
+
+  @Test
+  public void testTruncateLong() {
+    Assert.assertEquals(
+        "Should correctly truncate long values given a positive width and a long",
+        2147483647L,
+        TruncateUtil.truncateLong(Integer.MAX_VALUE, ((long) Integer.MAX_VALUE) + 1));
+  }
+
+  @Test
+  public void testTruncateDecimal() {
+    int width = 10;
+    Assert.assertEquals(
+        "Should return a decimal with the correct truncated value",
+        BigDecimal.valueOf(1230L, 2),
+        TruncateUtil.truncateDecimal(BigInteger.valueOf(width), BigDecimal.valueOf(1234L, 2)));
+
+    int scale = 2;
+    BigDecimal input = BigDecimal.valueOf(1234L, scale);
```

Comment:
Why use a local variable for `scale` here, but not above? It isn't referenced anywhere because the expected value is `input.scale()`

Reply:
Removed.